### PR TITLE
XSA-387 CVE-2021-28703

### PR DESCRIPTION
**Pull request by xen-project-security-team**
XSA-387 CVE-2021-28703

CVE data entry for:
  CVE-2021-28703

(This MR was automatically generated by the Xen Project Security Team's
xensec-internal-tools.  Please let us know if anything could be better.)

### CODE REVIEW

File: 2021/28xxx/CVE-2021-28703.json [hunk 1]
```diff
@@ -1,18 +1,84 @@
 {
```
Comment (cve-team) [inline 2021/28xxx/CVE-2021-28703.json:1]: 'affects' is a required property

**Review by cve-team — CHANGES_REQUESTED** [review]
Auto validation errors:

**Comment by xen-project-security-team** [comment]
CVE Team writes ("Re: [CVEProject/cvelist] XSA-387 CVE-2021-28703 (PR #3549)"):
...
> In 2021/28xxx/CVE-2021-28703.json:
> 
> > @@ -1,18 +1,84 @@
>  {
> 
> 'affects' is a required property

Hi.

We have no idea how to represent realitiy in your JSON schema I'm
afraid.

Could you please see our vulnerability text,
  https://xenbits.xen.org/xsa/advisory-387.html
and advise ?

Speaking personally (and therefore not on behalf of the Xen Project
Security Team), I am not convinced that making this field mandatory
adds enough value.  Representing reality in this field is often really
awkward, and is not necessary for the primary function of the CVE
database.

Extracts from our advisory below.

Ian.

VULNERABLE SYSTEMS
==================

All Xen branches up to and including 4.13 are vulnerable,
but only if the patches for XSA-378 have not been applied.

Xen versions 4.13.4, 4.14.x and 4.15.x are not affected.

Only x86 HMV and PVH guests permitted to use grant table version 2
interfaces can leverage this vulnerability.  x86 PV guests cannot
leverage this vulnerability.  On Arm, grant table v2 use is explicitly
unsupported.



RESOLUTION
==========

Applying the following patch resolves the issue:
  x86/p2m: don't assert that the passed in MFN matches for a remove

This patch was supplied with XSA-378, as one of 378's prerequisites.
The fix has already been applied to Xen stable branches as follows:

c65ea16dbcafbe4fe21693b18f8c2a3c5d14600e   in Xen 4.14.x, 4.15.x
f50fbddbae81fcccae56d27317bd71cc0e678ba2   in Xen 4.13.4
d44643199c96ac22491ae002d3bcd1c989b95ea4   in xen.git#stable-4.12

**Comment by anthnysingleton** [comment]
Hello @xen-project-security-team,

Thanks for reaching out for assistance. Is it possible to represent the information in the json file as follows:

```
"affects": {
    "vendor": {
        "vendor_data": [
            {
                "product": {
                    "product_data": [
                        {
                            "product_name": "Xen",
                            "version": {
                                "version_data": [
                                    {
                                        "version_affected": "!",
                                        "version_name": "Branch 4.13.4",
                                        "version_value": "4.13.4"
                                    },
                                    {
                                        "version_affected": "!",
                                        "version_name": "Branch 4.14.x",
                                        "version_value": "4.14.x"
                                    },
                                    {
                                        "version_affected": "!",
                                        "version_name": "Branch 4.15.x",
                                        "version_value": "4.15.x"
                                    },
                                    {
                                        "version_affected": "<=",
                                        "version_name": "Branch 4.13",
                                        "version_value": "4.13"
                                    }
                                ]
                            }
                        }
                    ]
                },
                "vendor_name": "Xen Project"
            }
        ]
    }
}
```
These versions were just pulled from the description provided. I do hear your concerns around reality of the listing and would like to point you to review [CVE JSON Schema version 5](https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/docs/CVE_JSON_5.0_bundled.json) as that may provide a solution to this issue in the future.

If this is not helpful please let me know or reach out to cna-coordinator@mitre.org for further assistance.

Thanks,
Anthony
CVE Content Team

**Comment by xen-project-security-team** [comment]
I will c&p what you have suggested into the JSON in this MR.

However, I do not think that this information is useful to anyone in this form.  No-one who has only the JSON information you propose can do anything useful with it even manually, let alone automatically.

Indeed, I think in many cases there is no way to represent the real-world situation.  And it is significant manual work to transform the information we do have into this supposedly-machine-readable but actually-useless format.

IMO it would be doing our users a better service to write "?" for all versions in this part of the JSON data, as routine practice.

**Comment by xen-project-security-team** [comment]
(I should have said: I, Ian Jackson, wrote that, with my Xen hat on.)

**Comment by xen-project-security-team** [comment]
@anthnysingleton Thanks for your suggestion, which I have incorporated.  See above.  I'm not sure how to retrigger your robot and re-review, or remove the needs-work label.    -Ian Jackson

**Pull request merged**